cli/tools/registry/mod.rs

```diff
@@ -292,22 +177,12 @@ fn print_diagnostics(diagnostics: Vec<String>) {
   }
 }
 
-async fn perform_publish(
-  http_client: &Arc<HttpClient>,
-  mut publish_order_graph: PublishOrderGraph,
-  mut prepared_package_by_name: HashMap<String, PreparedPublishPackage>,
+async fn get_auth_headers(
+  client: &reqwest::Client,
+  registry_url: String,
+  packages: Vec<Rc<PreparedPublishPackage>>,
```

Comment:
Is it necessary for `PreparedPublishPackage` to be in an `Rc`? I think this can just be `&[PreparedPublishPackage]`? Then we could avoid the clone on `package.tarball` when publishing, which could be somewhat sizable.

Reply:
`package.tarball` is `bytes::Bytes` which is cheap to clone.